teacher.rb

```diff
@@ -0,0 +1,11 @@
+require_relative './person'
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```

Comment:
- [ ] in order to make the constructor work you should pass the `age`, `name` and `parent_permission` in this method and pass them to `super` so it will be passed to the `Person` class when a student is created. you can have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#inheritance) for an example

Reply:
Hello @mikethreels
Thanks for your feedback,
I did the changes which you stated, you can check them now